igor-web/src/main/groovy/com/netflix/spinnaker/igor/scm/KeelManifestsController.groovy

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.igor.scm
+
+import com.fasterxml.jackson.databind.DeserializationFeature
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory
+import com.netflix.spinnaker.igor.scm.bitbucket.client.BitBucketMaster
+import com.netflix.spinnaker.igor.scm.github.client.GitHubMaster
+import com.netflix.spinnaker.igor.scm.gitlab.client.GitLabMaster
+import com.netflix.spinnaker.igor.scm.stash.client.StashMaster
+import groovy.util.logging.Slf4j
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+
+/**
+ * Exposes APIs to retrieve keel declarative manifests from source control repos.
+ */
+@RestController
+@Slf4j
+@RequestMapping("/keel")
+class KeelManifestsController {
+  @Autowired(required = false)
+  StashMaster stashMaster
+
+  @Autowired(required = false)
+  GitHubMaster gitHubMaster
+
+  @Autowired(required = false)
+  GitLabMaster gitLabMaster
+
+  @Autowired(required = false)
+  BitBucketMaster bitBucketMaster
+
+  ObjectMapper yamlMapper = new ObjectMapper(new YAMLFactory()).
+    disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES)
+
+  static final String KEEL_MANIFESTS_BASE_PATH = ".netflix/spinnaker"
+
+  // Note: we expose a restricted endpoint here (as opposed to a full pass-thru to the SCM APIs) to limit the exposure
+  // of accessing repo contents with igor's credentials
+
+  @GetMapping(path = '/{scmType}/{projectKey}/{repositorySlug}/manifests')
```

Comment:
I'd like for `keel` to be removed from all these function names

Reply:
suggestion - since I'm not sure that all of these path params are mandatory, should we use the same convention like the boost apis, meaning convert them to query params and add the key? example:
https://<>?repoSlug=some-app&projectKey=~spkr

Reply:
The path parameters are indeed all mandatory, but changing them to query parameters is something I also mentioned it in another comment because it makes the URI more readable. Will make that change.